File: app/src/main/java/com/mynet/kazekima/wsfighterscard/record/RecordDialogFragment.java
/*
 * Copyright (c) 2018 Ryuusuke Azuma All Rights Reserved.
 */

package com.mynet.kazekima.wsfighterscard.record;

import android.app.Dialog;
import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.DialogFragment;

import com.mynet.kazekima.wsfighterscard.R;

public class RecordDialogFragment extends DialogFragment {

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        Context context = getContext();
        if (context == null) {
            return super.onCreateDialog(savedInstanceState);
        }
        AlertDialog.Builder builder = new AlertDialog.Builder(context);
        builder.setMessage(R.string.dialog_record_fighters)
                .setPositiveButton(R.string.dialog_record_fighters_ok, new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int id) {
                        // OK
                    }
                })
                .setNegativeButton(R.string.dialog_record_fighters_cancel, new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int id) {
                        // User cancelled the dialog
                    }
                });
        // Create the AlertDialog object and return it
        return builder.create();
    }
}

File: android/src/main/kotlin/com/mynet/kazekima/wsfighterscard/record/RecordDialogFragment.kt
/*
 * Copyright (c) 2026 Ryuusuke Azuma All Rights Reserved.
 */

package com.mynet.kazekima.wsfighterscard.record

import android.app.Dialog
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import com.mynet.kazekima.wsfighterscard.R

class RecordDialogFragment : DialogFragment() {
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val context = context ?: return super.onCreateDialog(savedInstanceState)

        val builder = AlertDialog.Builder(context)
        builder.setMessage(R.string.dialog_record_fighters)
                .setPositiveButton(R.string.dialog_record_fighters_ok) { dialog, id ->
                    // OK
                }
                .setNegativeButton(R.string.dialog_record_fighters_cancel) { dialog, id ->
                    // User cancelled the dialog
                }
        // Create the AlertDialog object and return it
        return builder.create()
    }
}